examples/tv-casting-app/linux/main.cpp

```diff
@@ -21,18 +21,65 @@
 #include <controller/CHIPCommissionableNodeController.h>
 #include <platform/CHIPDeviceLayer.h>
 #include <platform/ConfigurationManager.h>
+#include <support/CHIPArgParser.hpp>
 #include <system/SystemLayer.h>
 #include <transport/raw/PeerAddress.h>
 
 using namespace chip;
 using namespace chip::Controller;
+using chip::ArgParser::HelpOptions;
+using chip::ArgParser::OptionDef;
+using chip::ArgParser::OptionSet;
 
-constexpr int kTvDeviceType                         = 35;
-constexpr uint16_t commissioningWindowTimeoutInSec  = 3 * 60;
-constexpr uint32_t commissionerDiscoveryTimeoutInMs = 5 * 1000;
+constexpr int kVideoPlayerDeviceType                 = 35;
+constexpr uint16_t kOptionDeviceType                 = 't';
+constexpr uint16_t kCommissioningWindowTimeoutInSec  = 3 * 60;
+constexpr uint32_t kcommissionerDiscoveryTimeoutInMs = 5 * 1000;
 
-CommissionableNodeController commissionableNodeController;
-chip::System::SocketWatchToken token;
+CommissionableNodeController gCommissionableNodeController;
+chip::System::SocketWatchToken gToken;
+Mdns::DiscoveryFilter gDiscoveryFilter = Mdns::DiscoveryFilter(Mdns::DiscoveryFilterType::kDeviceType, kVideoPlayerDeviceType);
+
+bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier, const char * aName, const char * aValue)
+{
+    bool retval = true;
+
+    switch (aIdentifier)
+    {
+    case kOptionDeviceType:
+        if (strcasecmp(aValue, "all") == 0)
+        {
+            gDiscoveryFilter = Mdns::DiscoveryFilter();
+        }
+        else
+        {
+            uint16_t deviceType = (uint16_t) strtol(aValue, nullptr, 10);
```

Comment:
Should probably return false if the value returned by `strtol` does not test true for `CanCastTo<uint16_t>`.